src/DIRAC/TransformationSystem/Agent/TaskManagerAgentBase.py

```diff
@@ -115,7 +115,12 @@ def finalize(self):
       self._logInfo("Wait for threads to get empty before terminating the agent (%d tasks)" %
                     len(self.transInThread))
       self.transInQueue = []
+      i = 0
       while self.transInThread:
+        i += 1
+        if i % 15 == 0:
+          self._logInfo("Still waiting for threads terminating the agent (%d tasks)" %
+                        len(self.transInThread))
```

Comment:
I have re-written all the threading logic in v7r3 (for this and other agents) and this is certainly going making conflicts. Since these are only monitoring changes, I would not add them here.

Reply:
Are you referring to just 5e40269? I'm happy to split that commit out but for the others I'd rather include most of these in v7r2 to make it easier to check the logs in Kibana rather than waiting a likely long time until v7r3 is released.

Reply:
Yes, I am only referring to https://github.com/DIRACGrid/DIRAC/commit/5e40269042dbacb877c3ef6704f17aea343894fe

Reply:
Done